src/dispatch.rs

```diff
@@ -68,15 +68,16 @@ impl AsyncDispatcher {
     }
 
     /// Waits for all the systems to finish
-    /// and executes thread local systems (if there
-    /// are any).
     pub fn wait(&mut self) {
         self.signal
             .take()
             .expect(ERR_NO_DISPATCH)
             .wait()
             .expect("The worker thread may have panicked");
+    }
```

Comment:
If that's the case, you're *doing it wrong*<sup>*TM*</sup> The dispatcher assumes all the resources are available to the running system, which isn't the case then.